src/test_l1_dsVideoPort.c

```diff
@@ -581,53 +572,45 @@ void test_l1_dsVideoPort_negative_dsIsVideoPortEnabled(void) {
- * |05|Compare the array values and make sure they are equal| | dsERR_NONE | The values must be equal |
- * |06|Call dsVideoPortTerm() - Terminate the video port system | | dsERR_NONE | Termination must be successful |
+ * |04|Compare the array values with the values for device respectively and make sure they are equal| | dsERR_NONE | The values must be equal |
+ * |05|Call dsVideoPortTerm() - Terminate the video port system | | dsERR_NONE | Termination must be successful |
  * 
  */
 void test_l1_dsVideoPort_positive_dsIsDisplayConnected(void) {
-	gTestID = 9;
-	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
-
-	dsError_t status;
-	intptr_t handle[NUM_OF_PORTS]={HANDLE_ARRAY_INIT};
-
-	bool isConnectedArray1[NUM_OF_PORTS];
-	bool isConnectedArray2[NUM_OF_PORTS];
-
-	// Step 01: Initialize video port system
-	status = dsVideoPortInit();
-	UT_ASSERT_EQUAL(status, dsERR_NONE);
-
-	// Step 02: Get the video port handle
-	for (int i = 0; i < NUM_OF_PORTS ; i++) {
-		status = dsGetVideoPort(kPorts[i].id.type, kPorts[i].id.index, &(handle[i]));
-		UT_ASSERT_EQUAL(status, dsERR_NONE);
-	}
-
-	// Step 03: Check if video port is connected
-	for (int i = 0; i < NUM_OF_PORTS ; i++) {
-		status = dsIsDisplayConnected(handle[i], &isConnectedArray1[i]);
-		UT_ASSERT_EQUAL(status, dsERR_NONE);
-	}
-
-	// Step 04: Repeat the check for connection status
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		status = dsIsDisplayConnected(handle[i], &isConnectedArray2[i]);
-		UT_ASSERT_EQUAL(status, dsERR_NONE);
-	}
-
-	// Step 05: Compare the array values
-	for (int i = 0; i < NUM_OF_PORTS ; i++) {
-		UT_ASSERT_EQUAL(isConnectedArray1[i], isConnectedArray2[i]);
-	}
-
-	// Step 06: Terminate the video port system
-	status = dsVideoPortTerm();
-	UT_ASSERT_EQUAL(status, dsERR_NONE);
-
-	UT_LOG("\n Out %s\n", __FUNCTION__); 
+    gTestID = 9;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status;
+    intptr_t handle = 0;
+
+    bool isConnected1 = false;
+
+    // Step 01: Initialize video port system
+    status = dsVideoPortInit();
+    UT_ASSERT_EQUAL_FATAL(status, dsERR_NONE);
+
+    // Step 02: Get the video port handle
+    for (int i = 0; i < gDSvideoPort_NumberOfPorts ; i++) {
+        status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle));
+        UT_ASSERT_EQUAL(status, dsERR_NONE);
+        // Step 03: Check if video port is connected
+        status = dsIsDisplayConnected(handle, &isConnected1);
+        UT_ASSERT_EQUAL(status, dsERR_NONE);
+        // Step 04: Compare the array values
+        bool isConnected;
+        if (gSourceType == 0) {
+            isConnected = true;
```

Comment:
we can directly do the assert UT_ASSERT_EQUAL(isConnected1, TRUE);
no need of extra variable.
you can take this in clean up activity.